src/index.ts

```diff
@@ -1,39 +1,57 @@
 import debug = require('debug')
-import flatten = require('array-flatten')
 
 const log = debug('compose-middleware')
 
 export type Next<T = void> = (err?: Error | null) => T
 export type RequestHandler <T, U, V = void> = (req: T, res: U, next: Next<V>) => V
-export type ErrorHandler <T, U, V = void> = (err: Error, req: T, res: U, next: Next<V>) => V
+export type ErrorHandler <T, U, V = void> = (err: Error | null, req: T, res: U, next: Next<V>) => V
 export type Middleware <T, U, V = void> = RequestHandler<T, U, V> | ErrorHandler<T, U, V>
 
-export type Handler <T, U, V = void> = Middleware<T, U, V> | flatten.NestedArray<Middleware<T, U, V>>
+/**
+ * Compose a variadic number of middlewares into a single middleware.
+ */
+export function compose <T, U, V = void> (
+  middleware?: RequestHandler<T, U, V>,
+  ...middlewares: Middleware<T, U, V>[]
+): RequestHandler<T, U, V>
+export function compose <T, U, V = void> (
+  middleware: ErrorHandler<T, U, V>,
+  ...middlewares: Middleware<T, U, V>[]
+): ErrorHandler<T, U, V>
+export function compose <T, U, V = void> (
+  middleware?: Middleware<T, U, V>,
+  ...middlewares: Middleware<T, U, V>[]
+): Middleware <T, U, V> {
+  if (!middleware) {
+    return successes<T, U, V>()
+  } else if (middleware.length === 3) {
```

Comment:
I typically avoid the `else` if it's unnecessary, personal coding preference.

Reply:
:+1: